airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftCopyDestination.java

```diff
@@ -165,13 +178,43 @@ protected void acceptTracked(AirbyteMessage message) throws Exception {
      */
     @Override
     protected void close(boolean hasFailed) throws Exception {
-      RedshiftCopier.closeAsOneTransaction(new ArrayList<>(tableNameToCopier.values()), hasFailed, null);
+      RedshiftCopier.closeAsOneTransaction(new ArrayList<>(streamNameToCopier.values()), hasFailed, redshiftDb);
     }
 
   }
 
-  public static void main(String[] args) throws Exception {
-    new IntegrationRunner(new RedshiftCopyDestination()).run(args);
+  public static class S3Config {
+
+    public String bucketName;
+    public String region;
+    public String accessKeyId;
+    public String secretAccessKey;
+
+    public S3Config(JsonNode config) {
+      this.bucketName = config.get("s3_bucket_name").asText();
+      this.region = config.get("s3_bucket_region").asText();
+      this.accessKeyId = config.get("access_key_id").asText();
+      this.secretAccessKey = config.get("secret_access_key").asText();
+    }
+
+    public static boolean isPresent(JsonNode config) {
```

Comment:
why not move this check to the constructor and throw an IllegalArgumentException or some other checked exception? this way we don't have to remember to call `isPresent`

Reply:
this function does validation and is also used by the `RedshiftDestination` to decide which underlying strategy to use.

moving this to the constructor works, however it feels strange to be switching on an exception instead of an explicit check. I prefer this form. again, will merge in for now and iterate on your outside if you feel strongly.